tuf/api/metadata.py

```diff
@@ -1326,21 +1418,31 @@ def __eq__(self, other: Any) -> bool:
             and self.terminating == other.terminating
             and self.paths == other.paths
             and self.path_hash_prefixes == other.path_hash_prefixes
+            and self.succinct_hash_info == other.succinct_hash_info
         )
 
     @classmethod
     def from_dict(cls, role_dict: Dict[str, Any]) -> "DelegatedRole":
         """Creates ``DelegatedRole`` object from its json/dict representation.
 
         Raises:
-            ValueError, KeyError: Invalid arguments.
+            ValueError, KeyError, TypeError: Invalid arguments.
         """
-        name = role_dict.pop("name")
+        name = role_dict.pop("name", None)
         keyids = role_dict.pop("keyids")
         threshold = role_dict.pop("threshold")
         terminating = role_dict.pop("terminating")
         paths = role_dict.pop("paths", None)
         path_hash_prefixes = role_dict.pop("path_hash_prefixes", None)
+        succinct_hash_dict = role_dict.pop("succinct_hash_delegations", None)
+        succinct_hash_info = None
+        if succinct_hash_dict is not None:
+            if not isinstance(succinct_hash_dict, Dict):
+                raise ValueError("succinct_hash_delegations must be a dict")
```

Comment:
maybe not needed: the constructor will fail if it's not a dict

Reply:
I added this as the error I got was an `AttributeError: 'str' object has no attribute 'pop'` ` and I thought that it made sense to give a more concrete message, but python developers will know what that message means, so it probably it doesn't make sense.
Will remove this check.